### ISSUE TITLE
Pending color is hard to distinguish from solved color

### ISSUE THREAD

**Issue by FinnLidbetter** [icpctools/icpctools#978]
The PENDING_COLOR is very hard to distinguish from the SOLVED_COLOR for colorblind folks (myself included) https://github.com/icpctools/icpctools/blob/f800d88f749a9c80249cab640b5e1909f538119c/PresContest/src/org/icpc/tools/presentation/contest/internal/ICPCColors.java#L12-L13

In issue #54 from 5 years ago it was suggested to make the color configurable.

As in interrim solution, would you consider just changing the existing pending color? Perhaps some shade of blue or purple that is not a similar shade to the existing SOLVED_COLOR nor FAILED_COLOR, e.g., I got this one from an accessible color pallet generator: #7f02ff

Alternatively, if you don't want to take that approach, but would accept a Pull Request to make the colors configurable, then I might be willing to work on this if I could be pointed in the right direction for where such configuration would be defined.

**Comment by niemela** [comment]
> Alternatively, if you don't want to take that approach, but would accept a Pull Request to make the colors configurable, then I might be willing to work on this if I could be pointed in the right direction for where such configuration would be defined.

Configurable sounds good. The natural way would be to have a command line option.

**Comment by FinnLidbetter** [comment]
Do you imagine anyone wanting to run this with a different configured color palette if the default is something that is generally agreed to be colorblind accessible (and looks ok to everyone else too)? Or do you see making it configurable as a way to more easily discover a suitable color palette?

**Comment by niemela** [comment]
> Do you imagine anyone wanting to run this with a different configured color palette if the default is something that is generally agreed to be colorblind accessible (and looks ok to everyone else too)?

No, if there is a *generally* agreed upon and liked color, we should just use that always.

Reaching that consensus is probably (?) much harder than implementing the color option :).

**Comment by FinnLidbetter** [comment]
Here are some options:

1. Current PENDING_COLOR = (230,230,0)
<img width="1440" alt="resolver-scoreboard-current" src="https://github.com/icpctools/icpctools/assets/15002920/e62e8bfa-0f50-4f36-ab04-83fb26939edb">



2. PENDING_COLOR = (66, 114, 245)
<img width="1440" alt="resolver-scoreboard-4272f5" src="https://github.com/icpctools/icpctools/assets/15002920/be17c138-ccf2-4507-9d58-e0f925f584ac">



3. PENDING_COLOR = (0, 0, 230)
<img width="1440" alt="resolver-scoreboard-0000e6" src="https://github.com/icpctools/icpctools/assets/15002920/ac7726b9-e54d-43b1-9b0f-6a0a6eaf7290">


If you want me to try something else, please provide feedback or suggest a color and I can put together some more options. I am not representative of all colorblind individuals, but the 2 blues that I suggest here are very distinct to me and there is also some distinction when it is grayscale.

**Comment by FinnLidbetter** [comment]
4. PENDING_COLOR = (58,109,182)
<img width="1440" alt="scoreboard-resolver-3a6db6" src="https://github.com/icpctools/icpctools/assets/15002920/722d442e-970f-4d7b-ac67-29506e48562b">

5. PENDING_COLOR = (92, 138, 221).  This is the value of `BLUE` here https://github.com/icpctools/icpctools/blob/f800d88f749a9c80249cab640b5e1909f538119c/PresContest/src/org/icpc/tools/presentation/contest/internal/ICPCColors.java#L8
<img width="1440" alt="resolver-scoreboard-5c8add" src="https://github.com/icpctools/icpctools/assets/15002920/cc45fb15-e9b1-4196-b261-12b70725182b">


Personally, I like 2 or 5 best. Maybe 5 since it is a color that is already defined in source. But I am happy with any of 2,3,4,5. They are all clearly distinguishable to me. If there are other colours that you or anyone would like me to try then I can do that. If we can agree on one, then I can submit a PR for it (and include the relevant screenshot(s)).

**Comment by deboer-tim** [comment]
Just for the record, I don't have a problem making these configurable, but the majority of contests would never get to configuring them anyway so I far prefer that the default colors work well for color blind and general aesthetics, which helps with consistency and people recognizing what the different colors mean too.

Any particular reason you went with all blue options? Just curious if you're using a particular CCS or use elsewhere swayed you.

3 is my least favourite (pending standing out is good, but that's too piercing) and would lean toward 5 for the same reasons listed. Calling in our in-house color blind expert @ashoos72 😉 to see if he wants to rule out or prefers any.

**Comment by niemela** [comment]
> Personally, I like 2 or 5 best

I agree, 2 and 5 are the best (of the ones you have shown). Although, 8 absolutely works as well.

> Any particular reason you went with all blue options? Just curious if you're using a particular CCS or use elsewhere swayed you.

I think it's pretty natural. There are not really that many options.

At least in our culture green means "yes", red means "no", and yellow is perceived as "in between" those. That also includes orange as in between yellow and red.

Black wouldn't really work in the default dark schema, and white wouldn't work in the light.

This basically leaves us with blue, purple, brown, pink, and grey (and maybe cyan I guess?),

Out of those blue and grey are both visually pleasing and semantically reasonable for "pending". Don't you agree?

...and I do think blue looks better than grey.

**Comment by ashoos72** [comment]
@FinnLidbetter I am also colorblind and have a hard time seeing the current yellow. I also like your options 2, 5, 7,8. Having said that these things may look good on a monitor, but we need to see what they look like on a projector. They don't tend to be as sharp or as easy to read when projected.

I like the idea of configurable, but then you will end up losing the uniformity of what the resolver looks like. I like consistency. Currently, we can "easily" look at a resolver and know what the colors stand for. You make it configurable and who knows what you will end up, then people will then want it fully extended allowing to select an entire color scheme for all the items on the resolver.

**Comment by FinnLidbetter** [comment]
It sounds like we are pretty much all in agreement for just improving the default pending color and not making it configurable.

Seeing what the color choices look like on a projector is important, but I do not have access to a projector. Do you have a suggestion for a path forward? 

Without seeing them on a projector, I am confident that 2,3,4,5,7,8 would each be an improvement on the status quo from an accessibility perspective. If we could get agreement on one to try then I could put up a PR for that color and we could get it merged and get feedback the next time that this is attempted in front of a large audience. Alternatively, if someone here has access to a projector we could at least get sample size 1 feedback from looking at these options that I have suggested before committing to one.

**Comment by ashoos72** [comment]
I have a projector at home and can test, just need the screens to project
and I can take pictures with my phone.

My caveat is my projector is old so not sure how it will compare, but as
you said sample size 1.

--Samir

**Comment by FinnLidbetter** [comment]
Ok that sounds good. Here is a zip folder of the screenshots that I took of the resolver for options 1--8 
[resolver_pending_color_options.zip](https://github.com/user-attachments/files/15506744/resolver_pending_color_options.zip)

If you want a zip of each of the Resolver builds that I ran with the different colour options, I could probably provide that too along with an event feed if you want to work with an actual Resolver instead of screenshots.

**Comment by ashoos72** [comment]
I have set up my projector and took two videos a wide shot of the entire screen and a close up of the changing cells. This is an older projector so it can only do 1024x768, take a look and let me know what you think. Some of the colors wash out the white text.

https://github.com/icpctools/icpctools/assets/51238378/f114a167-b27a-4bfa-ad2c-b3bbdf9f6163
https://github.com/icpctools/icpctools/assets/51238378/c730ebb6-f761-4dd2-b95b-132e105235ab

**Comment by nickygerritsen** [comment]
Seems most of the blue ones are quite ok?

**Comment by FinnLidbetter** [comment]
Thank you for testing those @ashoos72. I think option 2 looked best there. But I agree that most of the blues are ok. 

Does anyone prefer a different option or can we move forward with option 2? I.e., PENDING_COLOR = (66, 114, 245).

**Comment by ashoos72** [comment]
I agree the blues looked best, but I liked the darker option 3 better, but
2 or 3 is good for me.

Having said that, I like the original choice of "Yellow".

Did you explore other shades of Yellow ? As much as I'd like to be able to
see the "Pending" status better, I think the theme of red/yellow/green is
better (given we find a better shade). Just my 2 cents...

--Sam

**Comment by FinnLidbetter** [comment]
While 3 stands out, Tim commented above that it is a little too piercing and I can sympathise with that. 

I did try looking for another shade of yellow. But the brightest yellow I could try (255,255,0) was not distinct enough from the green to be an improvement on the status quo (230,230,0). At least for me. The only yellow-ish colours I could find that were sufficiently distinct were very pale, like option 6 (if you can even still call that yellow?). I think these pale yellow colours do not look good and they are still not as easy to distinguish from the greens as the blues are.

I think if we wanted to stick with yellow and address the colorblindness accessibility issues, we would also need to find an alternative green.

This is all made more difficult by the fact that none of these are just a single colour being used, but rather a range of colours
https://github.com/icpctools/icpctools/blob/f800d88f749a9c80249cab640b5e1909f538119c/PresContest/src/org/icpc/tools/presentation/contest/internal/ICPCColors.java#L28-L31
So I think it will be hard to find a green and yellow pair that works well while also reserving a darker green for first to solve.

**Comment by niemela** [comment]
> Did you explore other shades of Yellow ? As much as I'd like to be able to see the "Pending" status better, I think the theme of red/yellow/green is better (given we find a better shade). Just my 2 cents...

In addition to the colorblindness issue of yellow, there is also the fact that with scoring revolvers you would want to use yellow for "in between" scores, so everything on the red-to-green-via-yellow scale is already "taken". (See #989)

**Comment by FinnLidbetter** [comment]
I put up a pull request a couple of weeks ago with Option 2 from this discussion. Is there anything more I can do to help build consensus around this (or a different option)? Or does this just need someone with authority to accept it and merge it in?

**Comment by deboer-tim** [comment]
Sorry for the delay @FinnLidbetter , I've added a comment to the PR and merged. Thanks for your contribution!
